File: Medium/1318. Minimum Flips to Make a OR b Equal to c/MinimumFlipsToMakeaORbEqualToc.py
'''
LeetCode Medium: 1318. Minimum Flips to Make a OR b Equal to c
https://leetcode.com/problems/minimum-flips-to-make-a-or-b-equal-to-c/

Check MinimumFlipsToMakeaORbEqualToc.cpp for brief solution explanation.
'''
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        count = 0
        i = 0
        while((1<<i)<=max(a, max(b, c))):
            a_i = (1<<i)&a
            b_i = (1<<i)&b
            c_i = (1<<i)&c
            if c_i==0:
                if a_i>0:
                    count+=1
                if b_i>0:
                    count+=1
            else:
                if a_i+b_i==0:
                    count+=1
            i+=1
        return count

File: Medium/1318. Minimum Flips to Make a OR b Equal to c/MinimumFlipsToMakeaORbEqualToc.java
/*
LeetCode Medium: 1318. Minimum Flips to Make a OR b Equal to c
https://leetcode.com/problems/minimum-flips-to-make-a-or-b-equal-to-c/

Check MinimumFlipsToMakeaORbEqualToc.cpp for brief solution explanation.
*/
class Solution {
    public int minFlips(int a, int b, int c) {
        int count = 0;
        for(int i = 0;(1<<i)<=Math.max(a, Math.max(b, c));i++){
            int a_i = a&(1<<i);
            int b_i = b&(1<<i);
            int c_i = c&(1<<i);
            if(c_i==0){
                if(a_i>0) count++;
                if(b_i>0) count++;
            }else{
                //one of either a_i and b_i needs to be 1
                if(a_i+b_i==0){
                    count++;
                }
            }
        }
        return count;
    }
}